src/utransport.rs

```diff
@@ -16,9 +16,23 @@ use std::hash::{Hash, Hasher};
 use std::ops::Deref;
 use std::sync::Arc;
 
-use crate::{UMessage, UStatus, UUri};
+use crate::{UCode, UMessage, UStatus, UUri};
 
-/// `UListener` is the uP-L1 interface that provides a means to create listeners which are registered to [`UTransport`]
+/// A factory for URIs representing this uEntity's resources.
+///
+/// Implementations may use arbitrary mechanisms to determine the information that
+/// is necessary for creating URIs, e.g. environment variables, configuration files etc.
+pub trait LocalUriProvider: Send + Sync {
+    /// Gets the _authority_ used for URIs representing this uEntity's resources.
+    fn get_authority(&self) -> String;
+    /// Gets a URI that represents a given resource of this uEntity.
+    fn get_resource(&self, resource_id: u16) -> UUri;
```

Comment:
Naming `get_resource` misled me this will return `resource_id`. Since I guess the function is only used in Publish, how about `get_publish_uuri` or something similar?

Reply:
It will also be needed for registering an RPC server endpoint. What about `get_resource_uri`?

Reply:
Sound good to me.